plugins/jobs/src/js/components/form/VolumesFormSection.tsx

```diff
@@ -0,0 +1,220 @@
+import FormGroupHeadingContent from "#SRC/js/components/form/FormGroupHeadingContent";
+import FormRow from "#SRC/js/components/form/FormRow";
+import InfoTooltipIcon from "#SRC/js/components/form/InfoTooltipIcon";
+import MetadataStore from "#SRC/js/stores/MetadataStore";
+import {
+  FormOutput,
+  FormError,
+  JobVolume,
+  SecretVolume,
+  VolumeMode
+} from "./helpers/JobFormData";
+import { getFieldError } from "./helpers/ErrorUtil";
+
+interface VolumesSectionProps {
+  formData: FormOutput;
+  errors: FormError[];
+  showErrors: boolean;
+  onRemoveItem: (path: string, index: number) => void;
+  onAddItem: (path: string) => void;
+}
+
+function isFBS(volume: JobVolume | SecretVolume): boolean {
+  return volume.hasOwnProperty("secret");
+}
+
+class VolumesFormSection extends React.Component<VolumesSectionProps> {
+  constructor(props: VolumesSectionProps) {
+    super(props);
+  }
+
+  getVolumesLines() {
+    const {
+      formData: { volumes = [] },
+      onRemoveItem,
+      errors,
+      showErrors
+    } = this.props;
+
+    return volumes.map((volume: JobVolume | SecretVolume, index: number) => {
+      if (isFBS(volume)) {
+        return null;
+      }
+      let containerPathLabel = null;
+      let hostPathLabel = null;
+      let modeLabel = null;
+
+      let mode: any = (volume as JobVolume).mode;
+
+      if (
+        !(
+          (volume as JobVolume).mode === VolumeMode.RO ||
+          (volume as JobVolume).mode === VolumeMode.RW
+        )
+      ) {
+        // If mode is changed to something not in the list (for ex: in the JSON editor),
+        // show the default "Select ..." option
+        mode = "";
+      }
+
+      if (index === 0) {
+        containerPathLabel = (
+          <FieldLabel>
+            <FormGroupHeading>
+              <FormGroupHeadingContent primary={true}>
+                <Trans>Container Path</Trans>
+              </FormGroupHeadingContent>
+            </FormGroupHeading>
+          </FieldLabel>
+        );
+        hostPathLabel = (
+          <FieldLabel>
+            <FormGroupHeading>
+              <FormGroupHeadingContent primary={true}>
+                <Trans>Host Path</Trans>
+              </FormGroupHeadingContent>
+            </FormGroupHeading>
+          </FieldLabel>
+        );
+        modeLabel = (
+          <FieldLabel>
+            <FormGroupHeading>
+              <FormGroupHeadingContent primary={true}>
+                <Trans>Mode</Trans>
+              </FormGroupHeadingContent>
+            </FormGroupHeading>
+          </FieldLabel>
+        );
+      }
+
+      const containerPathErrors = getFieldError(
+        `job.run.volumes.${index}.containerPath`,
+        errors
+      );
+      const hostPathErrors = getFieldError(
+        `job.run.volumes.${index}.hostPath`,
+        errors
+      );
+      const modeErrors = getFieldError(`job.run.volumes.${index}.mode`, errors);
+
+      return (
+        <FormRow key={index}>
```

Comment:
index as key is probably insufficient since it's not permanent and unique. Is there another ID you can use?